lib/types/src/features.rs

```diff
@@ -387,6 +387,23 @@ impl Features {
 
         Ok(features)
     }
+
+    /// Checks if all needed features are supported by a specific compiler
+    pub fn supports_all_features(needed_features: &Features, compiler_features: &Features) -> bool {
```

Comment:
How is this different from `contains_features` ? Basically supports `supports_all_features(needed_features, compiler_features)` is the same as `compiler_features.contains_features(needed_features)`, isn't it?